File: 11743/main.c
#include <stdio.h>

int main(int argc, char* argv[])
{
   int t, i, j;
   int number, checksum;
   char input[32];

   scanf("%d\n", &t);
   while (t-- > 0) {
      gets(input);
      checksum = 0;
      for (i = j = 0; input[i] != '\0'; i++) {
         if (input[i] != ' ') {
            ++j;
            number = input[i] - '0';
            if (j % 2 == 1) {
               number += number;
               checksum += ((number / 10) + (number % 10));
            } else {
               checksum += number;
            }
         }
      }
      puts((checksum % 10 == 0) ? "Valid" : "Invalid");
   }

   return 0;
}

File: 11743/Main.java
import java.io.*;

public class Main {
   public static void main(String args[]) throws IOException {
      StringBuilder answer = new StringBuilder();
      BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
      int t = Integer.parseInt(in.readLine());

      while (t-- > 0) {
         int checksum = 0;
         String input = in.readLine();
         for (int i = 0, j = 0; i < input.length(); i++, j++) {
            if (input.charAt(i) == ' ') {
               --j;
               continue;
            }
            int n = input.charAt(i) - '0';
            if (j % 2 == 0) {
               n += n;
               checksum += ((n / 10) + (n % 10));
            } else {
               checksum += n;
            }
         }
         answer.append((checksum % 10 == 0) ? "Valid\n" : "Invalid\n");
      }
      System.out.print(answer);
   }
}
